### ISSUE TITLE
Agricraft villagers and greenhouses issue

### ISSUE THREAD

**Issue by Liiw** [InfinityRaider/AgriCraft#840]
Hello, 

I have installed Agricraft on 1.10.2 along with numerous other mods. Upon testing several singleplayer worlds, I found that the agricraft villager does not trade anything. The GUI shows up, but the icon squares are empty

I have also noticed that the greenhouses are all empty, except for patches of farmland, sometimes some irrigation systems and some patches of grass on the farmland.

Could this be a mod conflict issue? I am sorry if this is a duplicate issue, I searched, but could not find a similar issue beforehand.

**Comment by Nanianmichaels** [comment]
I can confirm this issue (I'm on a9 right now). In my testing, I've come across:

1) Ill-formed greenhouses (truncated generation, or pattern repetition that leads to partially-broken greenhouses, like an Irrigation Tank clipping a wall), though I've not seen missing irrigation channels.

2) No crops spawning inside the greenhouses. Not even cropsticks appear. The Seed Analyzer does spawn, though, as do the Irrigation Channels and Sprinklers, as stated before.

3) I couldn't get the trade GUI to appear for the villager. Also, its profession still appears as a long ".name" string.

I've also seen what may or may not be related issues: lack of loot chest filling on some structures. Vanilla structures all get proper loot in their chests, as do IE crates, but chests in RailCraft or RecurrentComplex buildings are always empty.

My wildy uneducated guess is that maybe one or more mods that tinker with village worldgen (which would mean Agricraft, RailCraft, RecurrentComplex, or Mo'Villages, most likely, since I have all of those installed at the moment) are conflicting with each other at worldgen time, and causing the lack of chest and crop population.

And potentially any mod that adds other villagers (including Agricraft, RC, Forestry, ExtraUtils, etc.) might be conflicting, too.

Liiw, which other mods that add village structures and/or villagers do you have installed? Maybe it's something we are able to at least track down, if it's a mod interaction issue.

**Comment by Liiw** [comment]
Hello again,

Your description of the problem is spot on! Thank you for elaborating. I have all the mods, you listed, installed as well as some others that (probably) add village generation. Unfortunately, I am not sure which mods add village generation since I really do have quite a number of them installed. I do believe that SGCraft adds a Tok'Ra villager. I have attached a list of mods I am currently using, maybe you or anyone else has a better idea which mod could be the culprit.

[list.txt](https://github.com/InfinityRaider/AgriCraft/files/785314/list.txt)

**Comment by Liiw** [comment]
Hello once again!

I started up a world and then searched the fml-client-latest log file for any mentions of  "farmer". I found this:   
        
[Agricraft villager warning.txt](https://github.com/InfinityRaider/AgriCraft/files/785342/Agricraft.villager.warning.txt)

My experience with Java is quite limited, but could this not indicate the issue?

**Comment by Nanianmichaels** [comment]
Apologies, busy day.

From my testing (check the edits), even an instance with just Pam's and Agricraft results in the issues described originally. I did add JourneyMap, to be able to look for villages easier, but I don't think that would affect worldgen, since that mod doesn't, AFAIK, tinker with worldgen.

Which means it's either an Agricraft issue, a Pam's issue, or a mod interaction issue between those two.

Unfortunately, I've disabled Agricaft on my instance, for now. I'd love it, but between Forestry, several ruins/exploration mods, and MineColonies, there's plenty of content to make me want to play for a while. No magic-based mods, though if TC ever releases for 1.10, I'll add it for sure.

Btw, I'm available for testing a fix to either/both of these issues.
